### PR TITLE
Mm/drawer code box style

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,4 +20,13 @@ clean_workspace:
...
 	@# Checks that Node is installed
-	@node -v > /dev/null 2>&1 || (echo -e 'node is not installed\n' && exit 42)
\ No newline at end of file
+	@node -v > /dev/null 2>&1 || (echo -e 'node is not installed\n' && exit 42)
+
+
+build_core: 
+	pushd packages/core && npm run build && popd
+
+build: build_core
```
Comment (Heyrows) [inline Makefile:29]: why do you need to write build_core at the end ?

File: packages/slice-machine/components/CodeBlock/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const text = lang
+    ? hljs.highlight(children, { language: lang }).value
+    : hljs.highlightAuto(children, DEFAULT_LANGUAGES).value;
+
+  return (
+    <Flex as="pre" style={style}>
+      <code
+        className="hljs"
+        style={{ overflowX: "auto", padding: "3px 5px" }}
+        dangerouslySetInnerHTML={{ __html: text }}
```
Comment (Heyrows) [inline packages/slice-machine/components/CodeBlock/index.tsx:29]: isn't that dangerous ?
It's one of the things that isn't recommended for XSS.

File: packages/slice-machine/package.json [hunk 3]
```diff
@@ -105,7 +106,6 @@
     "@testing-library/react": "^12.1.2",
     "@testing-library/react-hooks": "^7.0.0",
     "@theme-ui/color": "0.8.4",
-    "@theme-ui/prism": "0.8.4",
```
Comment (Heyrows) [inline packages/slice-machine/package.json:108]: why did we need to remove this ?

File: packages/slice-machine/src/theme.ts [hunk 4]
```diff
@@ -149,19 +150,19 @@ const AppTheme = (): Theme =>
...
-        my: 1,
-        color: "textClear",
-        bg: "grey01",
-        p: "2px",
-        overflow: "auto",
-        code: {
-          color: "inherit",
-        },
-      },
+      // pre: {
```
Comment (Heyrows) [inline packages/slice-machine/src/theme.ts]: if it's not use anymore please remove it :)

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import React, { useRef, useState } from "react";
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx:1]: do you think we could use only one CodeBlock component or are they too different ?

File: Makefile (hunk 1 above)
Comment (MarcMcIntosh) [reply Makefile:29]: That's a way to state that build relies on build_core :)

File: packages/slice-machine/components/CodeBlock/index.tsx (hunk 2 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/components/CodeBlock/index.tsx:29]: Yes it's dangerous, but most of the content is hard-coded in the hints a side from a few things like field id

File: packages/slice-machine/package.json (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/package.json:108]: It's not used anymore :)

File: packages/slice-machine/src/theme.ts (hunk 4 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/theme.ts]: will do :)

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx (hunk 5 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx:1]: The both have slightly different styles props.
and this CodeBlock has icons and a copy function :)

**Review by Miniplop — COMMENTED** [review]
Why there is modification in the lock files from all the packages in this PR, only the slicemachine-ui should be changed no ?

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ import React from "react";
 import { Flex, Text } from "theme-ui";
 
 import StepSection from "./components/StepSection";
+import CodeBlock from "../../../../components/CodeBlock";
```
Comment (Miniplop) [inline packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx]: components/Codeblocks

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx [hunk 7]
```diff
@@ -25,14 +26,10 @@ const NextSetupSteps: React.FunctionComponent<NextSetupStepProps> = ({
...
         </Text>
-        <Flex>
-          <pre
-            style={{ overflowX: "auto", padding: 8, border: "solid 1px grey" }}
-          >
-            npm install --save next-slicezone prismic-reactjs
-            @prismicio/slice-canvas-renderer-react
-          </pre>
-        </Flex>
+        <CodeBlock>
```
Comment (Miniplop) [inline packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx:29]: we don't specify the language ? How highlight js know which one to use ?

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NuxtSetupSteps.tsx [hunk 8]
```diff
@@ -3,6 +3,7 @@ import React from "react";
 import { Flex, Text } from "theme-ui";
 
 import StepSection from "./components/StepSection";
+import CodeBlock from "../../../../components/CodeBlock";
```
Comment (Miniplop) [inline packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NuxtSetupSteps.tsx]: components/Codeblock

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+import { useThemeUI, Text, Button, Flex, Box } from "theme-ui";
+
+import { BsCode } from "react-icons/bs";
+import { BiCopy } from "react-icons/bi";
+import { MdCheck } from "react-icons/md";
+import Code from "../../../../../../../components/CodeBlock";
```
Comment (Miniplop) [inline packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx]: components/codeBlocks

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+import type { Language } from "../../../../../../../components/CodeBlock";
```
Comment (Miniplop) [inline packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx]: Maybe you can use one import for both no ?

File: packages/slice-machine/tests/project/package.json [hunk 11]
```diff
@@ -29,6 +29,7 @@
   },
   "dependencies": {
     "@prismicio/vue": "^2.0.7",
+    "nuxt": "^2.15.8",
```
Comment (Miniplop) [inline packages/slice-machine/tests/project/package.json]: why for ?

File: packages/slice-machine/src/theme.ts (hunk 4 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/theme.ts]: done :)

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx (hunk 7 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx:29]: There's an option to specify which one, it's the same result.
I guess it tries a few languages and see which one works best :)

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx (hunk 10 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx]: probally

File: packages/slice-machine/tests/project/package.json (hunk 11 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/tests/project/package.json]: so I could trick the ui to thinking either next or nuxt was being used and I could see the set-up drawer :)

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx (hunk 9 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx]: done

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NuxtSetupSteps.tsx (hunk 8 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NuxtSetupSteps.tsx]: done

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx (hunk 6 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/NextSetupSteps.tsx]: done

**Review by Heyrows — APPROVED** [review]

File: packages/slice-machine/tests/project/package.json (hunk 11 above)
Comment (Miniplop) [reply packages/slice-machine/tests/project/package.json]: Oh to do so, you can put the framework in the sm.json

Comment (MarcMcIntosh) [reply packages/slice-machine/tests/project/package.json]: done :)

File: packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx (hunk 10 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/common/Zone/Card/components/Hints/CodeBlock.tsx]: done

**Review by Miniplop — APPROVED** [review]

**Pull request merged**